BUG: metakernel API error out on leapseconds kernel

Error message
```
[ERROR] SpiceNOLEAPSECONDS: 
================================================================================

Toolkit version: CSPICE_N0067

SPICE(NOLEAPSECONDS) --

The variable that points to the leapseconds (DELTET/DELTA_AT) could not be located in the kernel pool.  It is likely that the leapseconds kernel has not been loaded.

utc2et_c --> UTC2ET --> TTRANS

================================================================================
Traceback (most recent call last):
  File "/var/task/SDSCode/api_lambdas/spice_metakernel_api.py", line 206, in lambda_handler
    start_time, end_time = _convert_input_times_to_j2000(start_time_str, end_time_str)
  File "/var/task/SDSCode/api_lambdas/spice_metakernel_api.py", line 178, in _convert_input_times_to_j2000
    start_date = spiceypy.datetime2et(start_date_datetime)
  File "/opt/python/spiceypy/spiceypy.py", line 139, in with_errcheck
    check_for_spice_error(f)
  File "/opt/python/spiceypy/spiceypy.py", line 122, in check_for_spice_error
    raise dynamically_instantiate_spiceyerror(
```

Reproduce the bug using this command - https://api.imap-mission.com/metakernel?start_time=20260101&end_time=20260404&file_types=ephemeris_predicted

--- a/sds_data_manager/lambda_code/SDSCode/spice_utilities.py
+++ b/sds_data_manager/lambda_code/SDSCode/spice_utilities.py
@@ -88,7 +88,7 @@ def furnish_best_spice_file(kernel_type: str):
         or the file is not in the S3 bucket, FileNotFoundError will raise.
     """
     # Check if S3_BUCKET and DATA_DIR are set
-    if "S3_BUCKET" not in os.environ or "DATA_DIR" not in os.environ:
+    if "S3_BUCKET" not in os.environ or "DATA_DIR" not in imap_data_access.config:
         raise FileNotFoundError(
             f"Unable to find the latest {kernel_type} kernel. "
             "Please ensure S3_BUCKET and DATA_DIR are set in the environment variables."

--- a/tests/lambda_endpoints/test_spice_metakernel_api.py
+++ b/tests/lambda_endpoints/test_spice_metakernel_api.py
@@ -8,6 +8,9 @@ import imap_data_access
 
 from sds_data_manager.lambda_code.SDSCode.api_lambdas import spice_metakernel_api
 from sds_data_manager.lambda_code.SDSCode.database import models
+from sds_data_manager.lambda_code.SDSCode.spice_utilities import (
+    MAXIMUM_MISSION_J2000_TIME,
+)
 
 
 def _irrelevant_data():
@@ -339,3 +342,62 @@ def test_metakernel_frames(session):
         None,
     )
     assert len(json.loads(result["body"])) == 2
+
+
+def test_metakernel_with_human_readable_dates(session, s3_client):
+    """Test metakernel with human readable dates as input."""
+    # Upload time kernel to S3 so furnish_best_spice_file can find it
+    current_path = os.path.dirname(os.path.abspath(__file__))
+    one_level_up = os.path.abspath(os.path.join(current_path, ".."))
+    test_spice_data_dir = os.path.join(one_level_up, "test-data", "test_spice_files")
+
+    lsk_test_path = os.path.join(test_spice_data_dir, "naif0012.tls")
+    bucket_name = os.getenv("S3_BUCKET")
+    with open(lsk_test_path, "rb") as f:
+        s3_client.put_object(
+            Bucket=bucket_name,
+            Key="imap/spice/lsk/naif0012.tls",
+            Body=f,
+        )
+
+    # Insert time kernel into db so furnish_best_spice_file can find it.
+    _insert_test_file(
+        session, "naif0012.tls", [[1, MAXIMUM_MISSION_J2000_TIME]], upload_time=1
+    )
+    # Insert attitude file for testing file_types parameter
+    _insert_test_file(
+        session, "imap_1000_001_1000_100_002.ah.bc", [[1, 300]], upload_time=1
+    )
+
+    # Attempt to query with human-readable date format without specifying file_types
+    result = spice_metakernel_api.lambda_handler(
+        {
+            "queryStringParameters": {
+                "start_time": "20260101",
+                "end_time": "20260401",
+                "list_files": "True",
+            }
+        },
+        None,
+    )
+    # Should find leapseconds kernel and return successfully
+    assert result["statusCode"] == 200
+    assert json.loads(result["body"]) == ["naif0012.tls"]
+
+    # Now query with file_types specified - should succeed
+    result = spice_metakernel_api.lambda_handler(
+        {
+            "queryStringParameters": {
+                "start_time": "19000101",
+                "end_time": "20260101",
+                "spice_path": "",
+                "list_files": "True",
+                "file_types": "attitude_history",
+            }
+        },
+        None,
+    )
+    # Should return successfully with specific kernel available
+    assert result["statusCode"] == 200
+    # Should return the attitude file that was inserted
+    assert json.loads(result["body"]) == ["imap_1000_001_1000_100_002.ah.bc"]

